File: SocketComm.Examples/SendValue/FloatServer.java
import java.nio.ByteBuffer;

public class FloatServer {
    public static void main(String[] args) throws Exception {
        FPipe p = new FPipe("/tmp/floatvalue.pipe");
        p.ExpectCmd(Cmd.Ready);
        float i = 1.1f;
        while (i < Float.MAX_VALUE && i > 1)
        {
            System.out.printf("Sending %f\n", i);
            p.SendValue(i, (val)-> ByteBuffer.allocate(4).putFloat(val));
            p.ExpectCmd(Cmd.Ready);
            p.WriteCmd(Cmd.Go);
            p.ExpectCmd(Cmd.Done);
            p.ExpectCmd(Cmd.Receive);
            i = p.ReceiveValue(ByteBuffer::getFloat);
            System.out.printf("Received %f\n", i);
        }
        p.WriteCmd(Cmd.Exit);
        p.close();
    }
}

File: SocketComm.Examples/SendValue/FloatServer.cs
using static SocketComm.Cmd;
namespace SocketComm.Examples.SendValue;

public class FloatServer
{
    public static void Main()
    {
        FPipe p = new FPipe("/tmp/floatvalue.pipe");
        p.Connect();
        try
        {
            p.ExpectCmd(Ready);
            float i = 1.1f;
            while (i is < float.MaxValue and > 1)
            {
                Console.WriteLine($"Sending {i}");
                p.SendValue(i, SimpleConversion.NumberToBytes);
                p.ExpectCmd(Ready);
                p.WriteCmd(Go);
                p.ExpectCmd(Done);
                p.ExpectCmd(Receive);
                i = p.ReceiveValue(SimpleConversion.BytesToNumber<float>);
                Console.WriteLine($"Received {i}");
            }
        }
        catch (Exception e)
        {
            Console.Error.WriteLine("An error occured with the pipe: " + e);
        }
        finally
        {
            p.WriteCmd(Exit);
            p.Dispose();
        }
        
    }
}